### PR TITLE
Iss 214

### PR DESCRIPTION
**Pull request by joeyzhu00**
Vehicle heading taken from using GPS measurements combined with the IMU rate of rotation in the EKF. A program to convert the `Odometry` message from the EKF to an `Imu` message has been included for lidar mapping experimentation.

A separate issue will be made after discussing a method for retrieving an initial vehicle heading without the use of GPS (due to the heading estimate's need for vehicle movement).

### CODE REVIEW

File: src/modules/localization/src/frame_tf_calibrate/frame_calibration.cpp [hunk 1]
```diff
@@ -193,18 +169,77 @@ namespace frame_calibration_node
...
                 else
                 {
-                    // calculate the euclidian distance between point (n) and point (n-1)
                     float distance_comparison;
                     float gps_threshold;
+                    float vehicle_heading_from_gps;
+                    float lpf_coeff;
+                    int sample_size;
 
+                    frameCalibrationNode_nh.getParam("/frame_calibration/lpf_coeff", lpf_coeff);
```
Comment (drofp) [inline src/modules/localization/src/frame_tf_calibrate/frame_calibration.cpp]: Can you comment this please?

File: src/modules/localization/src/frame_tf_calibrate/frame_calibration.cpp [hunk 2]
```diff
@@ -193,18 +169,77 @@ namespace frame_calibration_node
...
                     
+                    // calculating vehicle heading from gps point n and gps point n-1
+                    vehicle_heading_from_gps = std::atan2((geodesy_msg.pose.pose.position.y - previous_geodesy_point.y), (geodesy_msg.pose.pose.position.x - previous_geodesy_point.x));
+
+                    // applying a low-pass filter and creating quaternion message to stuff into the geodesy_tf message
+                    if(second_message_sent)
+                    {
+                        vehicle_heading_from_gps = lpf_coeff * vehicle_heading_from_gps + (1 - lpf_coeff) * previous_vehicle_heading;
+
+                        // heading_deque.push_back(vehicle_heading_from_gps);
```
Comment (drofp) [inline src/modules/localization/src/frame_tf_calibrate/frame_calibration.cpp]: Consider making this it's own function

Comment (joeyzhu00) [reply src/modules/localization/src/frame_tf_calibrate/frame_calibration.cpp]: I'm just going to remove the low-pass filter and median filter functionality from the code. We most likely won't use either of the two filters as we can just depend on the EKF.

**Review by drofp — APPROVED** [review]
Builds fine

**Pull request merged**